src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -343,86 +341,112 @@ func (r *ClusterReconciler) WithClusterDomain(
 	return r
 }
 
+// nolint:funlen,gocyclo // External nodes list should be refactored
 func (r *ClusterReconciler) createExternalNodesList(
 	ctx context.Context,
 	pods []corev1.Pod,
 	pandaCluster *redpandav1alpha1.Cluster,
 	nodePortName types.NamespacedName,
-) (
-	external, externalAdmin, externalProxy []string,
-	proxyIngress string,
-	err error,
-) {
+) (*redpandav1alpha1.NodesList, error) {
 	externalKafkaListener := pandaCluster.ExternalListener()
 	externalAdminListener := pandaCluster.AdminAPIExternal()
 	externalProxyListener := pandaCluster.PandaproxyAPIExternal()
-	if externalKafkaListener == nil && externalAdminListener == nil {
-		return []string{}, []string{}, []string{}, "", nil
+	schemaRegistryConf := pandaCluster.Spec.Configuration.SchemaRegistry
+	if externalKafkaListener == nil && externalAdminListener == nil && externalProxyListener == nil &&
+		schemaRegistryConf == nil {
```

Comment:
shouldn't you check here rather that external connectivity is enabled for schema registry?

Reply:
Ok